core/src/test/java/org/apache/iceberg/hadoop/TestCatalogUtilDropTable.java

```diff
@@ -201,6 +188,25 @@ public void shouldNotDropDataFilesIfGcNotEnabled() {
         .containsAll(metadataLocations);
   }
 
+  private static FileIO createMockFileIO(FileIO wrapped) {
```

Comment:
These tests needed the mock object to support the added `newInputFile` methods. Rather than continuing to change each test individually, I refactored to have one method that is used throughout to create the mock `FileIO`.